pkg/eventing/v1/client.go

```diff
@@ -137,6 +143,31 @@ func (c *knEventingClient) UpdateTrigger(ctx context.Context, trigger *eventingv
 	return nil
 }
 
+func (c *knEventingClient) UpdateTriggerWithRetry(ctx context.Context, name string, updateFunc TriggerUpdateFunc, nrRetries int) error {
+	return updateTriggerWithRetry(ctx, c, name, updateFunc, nrRetries)
+}
+
+func updateTriggerWithRetry(ctx context.Context, c KnEventingClient, name string, updateFunc TriggerUpdateFunc, nrRetries int) error {
+	b := retry.DefaultRetry
+	b.Steps = nrRetries
+	err := retry.RetryOnConflict(b, func() error {
```

Comment:
If you have such a long inner function, for better readability, I'd suggest to move the function to a top-level, named function. If you need some closures arguments, you could pass it as argument, like e.g. in

```golang
err := retry.RetryOnConflict(b, func() error { 
      return myInnerFunc(closureArg);
}
```

If no closure is needed, you could just use the function name

```golang
err := retry.RetryOnConflict(b, myInnerFunc);
```